### ISSUE TITLE
Unable to watch 3D movies with external subtitles

### ISSUE THREAD

**Issue by markonjezic** [UniversalMediaServer/UniversalMediaServer#893]
Starting with version 6.3.0, I'm no longer able to watch any 3D movie that contains external subtitles. TV shows the error that the resource is not available when attempting to start playback. According to the logs RequestHandlerV2 stops with null pointer exception. Here's an excerpt from debug log:

```
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Opened request handler on socket /10.10.10.127:60370
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Request: HTTP/1.1 : HEAD : get/49/TEST.3D.HSBS.mp4_transcoded_to.mpg
DEBUG 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.configuration.RendererConfiguration Matched media renderer "BRAVIA KDL-55W815B" based on address /10.10.10.127
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Received on socket: getcontentFeatures.dlna.org: 1
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Received on socket: Pragma: getIfoFileURI.dlna.org
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Received on socket: transferMode.dlna.org: Streaming
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Received on socket: X-AV-Physical-Unit-Info: pa="BRAVIA KDL-55W815B";
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Received on socket: X-AV-Client-Info: av=5.0; cn="Sony Corporation"; mn="BRAVIA KDL-55W815B"; mv="1.7";
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Received on socket: Host: 10.10.11.11:5001
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Recognized media renderer: BRAVIA KDL-55W815B
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 HTTP: get/49/TEST.3D.HSBS.mp4_transcoded_to.mpg / 0-0
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.dlna.DLNAResource Asked stream chunk : TimeRange [start=null, end=null] of TEST.3D.HSBS.mp4 and player FFmpeg
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.dlna.DLNAResource Setting lastStartPosition from time-seeking: 0.0
DEBUG 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.dlna.DLNAResource Starting transcode/remux of TEST.3D.HSBS.mp4 with media info: container: mp4, bitrate: 2110591, size: 1500337473, video tracks: 1, audio tracks: 1, subtitle tracks: 1, video codec: h264, duration: 01:34:46.00, width: 1920, height: 1080, frame rate: 23.976, thumb size: 10539, mime type: video/mp4, attached fonts: false
	Audio track id: 0, lang: und, audio codec: AAC, sample frequency:44100, number of channels: 2, bits per sample: 16
	Subtitle track id: 100, type: SUBRIP, lang: cs, externalFile: F:\TEST\TEST.3D.HSBS.srt, external file character set: WINDOWS-1250
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Looking for an audio track with lang: eng
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Looking for an audio track with lang: fre
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Looking for an audio track with lang: jpn
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Looking for an audio track with lang: ger
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Looking for an audio track with lang: und
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Matched audio track: id: 0, lang: und, audio codec: AAC, sample frequency:44100, number of channels: 2, bits per sample: 16
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Searching for a match for: und with eng and off
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Searching for a match for: und with * and eng
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Searching for a match for: und with * and und
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.encoders.Player Matched external subtitles track that did not match language preferences: id: 100, type: SUBRIP, lang: cs, externalFile: F:\TEST\TEST.3D.HSBS.srt, external file character set: WINDOWS-1250
DEBUG 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.configuration.ConfigurationReader Reading ffmpeg_fontconfig: true (default: false)
DEBUG 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 Caught exception: null
TRACE 2016-05-11 14:35:48.223 [New I/O worker #4] net.pms.network.RequestHandlerV2 
java.lang.NullPointerException: null
	at net.pms.util.SubtitleUtils.getSubtitles(SubtitleUtils.java:172)
	at net.pms.encoders.FFMpegVideo.getVideoFilterOptions(FFMpegVideo.java:189)
	at net.pms.encoders.FFMpegVideo.launchTranscode(FFMpegVideo.java:942)
	at net.pms.dlna.DLNAResource.getInputStream(DLNAResource.java:2987)
	at net.pms.network.RequestV2.answer(RequestV2.java:374)
	at net.pms.network.RequestHandlerV2.writeResponse(RequestHandlerV2.java:300)
	at net.pms.network.RequestHandlerV2.messageReceived(RequestHandlerV2.java:256)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.handler.stream.ChunkedWriteHandler.handleUpstream(ChunkedWriteHandler.java:142)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.handler.codec.http.HttpChunkAggregator.messageReceived(HttpChunkAggregator.java:145)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:459)
	at org.jboss.netty.handler.codec.replay.ReplayingDecoder.callDecode(ReplayingDecoder.java:536)
	at org.jboss.netty.handler.codec.replay.ReplayingDecoder.messageReceived(ReplayingDecoder.java:435)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:559)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:268)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:255)
	at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:88)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108)
	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89)
	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178)
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108)
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by valib** [comment]
@markonjezic please send the full debug log.

**Comment by markonjezic** [comment]
@valib You can download the full debug log from here: http://www.max.rs/temp/ums_dbg.zip

**Comment by valib** [comment]
@markonjezic thanks for reporting. Fixed in 1c89f2b
